src/main/java/com/codessquad/qna/controller/UserController.java

```diff
@@ -0,0 +1,75 @@
+package com.codessquad.qna.controller;
+
+import com.codessquad.qna.repository.User;
+import com.codessquad.qna.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Controller
+public class UserController {
+
+    private List<User> users = new ArrayList<>();
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @GetMapping("/user/create")
+    public String toCreateUser() {
+        return "signup";
+    }
+
+    @PostMapping("/user/create")
+    public String createUser(User user) {
+        users.add(user);
+        userRepository.save(user);
+        return "redirect:/users";
+    }
+
+    @GetMapping("/users")
+    public String showUser(Model model) {
+        model.addAttribute("users", users);
+        return "list";
+    }
+
+    @GetMapping("/users/{userId}")
+    public String showProfile(@PathVariable("userId") String userId, Model model) {
+        for (User user : users) {
+            if (user.getUserId().equals(userId)) {
+                model.addAttribute("user", user);
+                break;
+            }
+        }
+        return "profile";
+    }
+
+    @GetMapping("/users/{userId}/updateForm")
+    public String updateForm(@PathVariable("userId") String userId, Model model) {
+        for (User user : users) {
+            if (user.isSameId(userId)) {
+                model.addAttribute("user", user);
+                break;
+            }
+        }
+        return "updateForm";
+    }
+
+    @PostMapping("/user/{user.userId}/update")
+    public String updateUser(@PathVariable("user.userId") String userId, User user) {
+        for (User temp : users) {
```

Comment:
```suggestion
        for (User user : users) {
```

아무렇게나 temp라는 단어를 남용하지 맙시다!

Reply:
해당 반복문에만 사용되는 임시변수명이라 일부러 temp로 지었습니다. 이전에 간소화된 for반복문에 temp를 임시변수로 사용하는 걸 많이 봤었는데 이게 잘못된 사용법인 건가요? 🤔

Reply:
해당 반복문에만 사용된다 하더라도, `temp`라는 단어가 적절하지는 않습니다.
임시로 데이터를 저장한다거나 하는 경우에만 제한적으로 `temp`를 써주도록 하는 편이 좋은 네이밍을 하는데 도움이 됩니다.

Reply:
그렇군요! 해당 부분 수정하고, 앞으로 `temp`는 최대한 덜 사용하겠습니다. 감사합니다🙇‍♀️!!